src/Engines/thinEngine.ts

```diff
@@ -1779,7 +1779,16 @@ export class ThinEngine {
             var order = effect.getAttributeLocation(index);
 
             if (order >= 0) {
-                var vertexBuffer = vertexBuffers[attributes[index]];
+                var ai = attributes[index];
+                var vertexBuffer: Nullable<VertexBuffer> = null;
+
+                if (extraVertexBuffers) {
+                    vertexBuffer = extraVertexBuffers[ai];

```

Comment:
Here I decided that if the user adds a non-instanced VertexBuffer on the Mesh it will override the Geometry VertexBuffer of the same kind. That can come in handy if you want a one-off different UV-set or something.

Reply:
Good for me!